lib/datadog/statsd/serialization/tag_serializer.rb

```diff
@@ -13,18 +13,21 @@ def initialize(global_tags = [], env = ENV)
 
           # Convert to tag list and set
           @global_tags = to_tags_list(global_tags)
+          @global_tags_formatted = @global_tags.join(',') if @global_tags.any?
         end
 
         def format(message_tags)
```

Comment:
I remember doing something in this file ages ago, I think because Statsd was returning some format that we couldn't use downstream in the tracing library. I think we wanted to merge tags at the tracer library level into the global stats tags?

I would double check the history on this, and make sure it doesn't break tags downstream.

Reply:
I searched `ddtrace`s history, and we don't access these global tags.

Regarding this in this repo, there are no modification attempts to global tags.

One thing that is technically possible today is to get a reference to it and mutate it:
```ruby
stats = Datadog::Statsd.new
stats.tags << 'new:tag' # stats.tags is effectively +TagSerializer#global_tags+.
```
Instead of using the public interface:
```ruby
stats = Datadog::Statsd.new(tags: ['new:tag'])
```

I'm not too convinced this should be supported behaviour by this library.

My suggestion is to freeze the internally stored `TagSerializer#global_tags`, thus having `TagSerializer#global_tags` guaranteed to return a read-only object.